spec/lib/twingly/http_spec.rb

```diff
@@ -605,4 +605,14 @@ class CustomError < StandardError; end
       end
     end
   end
+
+  describe "#initialize" do
+    context "when no logger is given" do
```

Comment:
This is, in lack of better words, a bit underspecified. What we **really** want to test is that if we don't provide a Logger, a Logger will be chosen for us. That given logger is also a kind of logger that logs stuff to `/dev/null`.

I would also introduce this new spec above the shared examples, I think it's better for readability (on that subject, the spec file is way too long, but hardly something we should address now :))

Reply:
> I would also introduce this new spec above the shared examples,

Sorry, just below the shared examples, but before `#post`

Reply:
What could be done here is to expose the `@logger` with an `attr_accessor`, cause then we could change the logger on the fly rather than having to re-instantiate the client. It would also be way easier to write a spec

Reply:
All right, now `#initialize` is moved to before `#post` and the examples have been updated to test that we were actually given the expected `Logger`. The latter was made possible because of the now exposed `@logger`.

(2ba0db1)